core/src/main/scala/org/apache/spark/util/JavaModuleUtils.scala

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.util
+
+import scala.collection.mutable
+
+import org.apache.commons.lang3.{JavaVersion, SystemUtils}
+
+import org.apache.spark.SparkConf
+import org.apache.spark.internal.config.{DRIVER_JAVA_OPTIONS, EXECUTOR_JAVA_OPTIONS, OptionalConfigEntry}
+
+object JavaModuleUtils {
+
+  private val javaModuleOptions = Set("--add-opens java.base/java.nio=ALL-UNNAMED",
+    "--add-opens java.base/sun.nio.ch=ALL-UNNAMED",
+    "--add-opens java.base/java.lang.invoke=ALL-UNNAMED",
+    "--add-opens java.base/java.nio=ALL-UNNAMED",
+    "--add-opens java.base/sun.nio.ch=ALL-UNNAMED",
+    "--add-opens java.base/java.lang.invoke=ALL-UNNAMED",
+    "--add-opens java.base/java.util=ALL-UNNAMED",
+    "--add-opens java.base/sun.security.action=ALL-UNNAMED",
+    "--add-opens java.base/sun.util.calendar=ALL-UNNAMED",
+    "--add-opens java.base/java.lang=ALL-UNNAMED",
+    "--add-opens java.base/sun.nio.cs=ALL-UNNAMED",
+    "--add-opens java.base/java.net=ALL-UNNAMED",
+    "--add-opens java.base/java.io=ALL-UNNAMED",
+    "--add-opens java.base/java.util.concurrent=ALL-UNNAMED",
+    "--add-exports java.base/jdk.internal.util.random=ALL-UNNAMED")
```

Comment:
Thank you, @LuciferYang !

Reply:
BTW, I didn't realized that we need `--add-exports`. Could you explain a little more about this?

Reply:
`--add-exports` is useless for the current pr ,  already remove it.

But It is needed in modules `mllib-local` and `mllib`, some code in test like `mock[java.util.Random]` need this `--add-exports`